packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -28,17 +28,31 @@ const getNextCellIndexes = (code: string, indexes: GridCellIndexCoordinates) =>
     throw new Error('Material-UI: The first argument (code) should be an arrow key code.');
   }
 
-  if (code === 'ArrowLeft') {
-    return { ...indexes, colIndex: indexes.colIndex - 1 };
-  }
-  if (code === 'ArrowRight') {
-    return { ...indexes, colIndex: indexes.colIndex + 1 };
-  }
-  if (code === 'ArrowUp') {
-    return { ...indexes, rowIndex: indexes.rowIndex - 1 };
+  switch (code) {
+    case 'ArrowLeft':
+      if (indexes.rowIndex === null) {
+        return { rowIndex: null, colIndex: indexes.colIndex - 1 };
+      }
+      return { ...indexes, colIndex: indexes.colIndex - 1 };
```

Comment:
no need refactoring here. if there is no rowIndex then leave it

Reply:
If there is no `rowIndex` then it will be `undefined` which is a problem. I think it should still have a value just be `null`.

Reply:
Updated